zipline/pipeline/pipeline.py

```diff
@@ -148,24 +148,57 @@ def set_screen(self, screen, overwrite=False):
             )
         self._screen = screen
 
-    def to_graph(self, screen_name, default_screen):
+    def to_execution_plan(self,
+                          screen_name,
+                          default_screen,
+                          all_dates,
+                          start_date,
+                          end_date):
         """
-        Compile into a TermGraph.
+        Compile into an ExecutionPlan.
 
         Parameters
         ----------
         screen_name : str
             Name to supply for self.screen.
         default_screen : zipline.pipeline.term.Term
             Term to use as a screen if self.screen is None.
+        all_dates : pd.DatetimeIndex
+            A calendar of dates to use to calculate starts and ends for each
+            term.
+        start_date : pd.Timestamp
+            The first date of requested output.
+        end_date : pd.Timestamp
+            The last date of requested output.
         """
+        return ExecutionPlan(
+            self._prepare_graph_terms(screen_name, default_screen),
+            all_dates,
+            start_date,
+            end_date,
+        )
+
+    def to_simple_graph(self, screen_name, default_screen):
+        """
+        Compile into a simple TermGraph with no extra row metadata.
+
+        Parameters
+        ----------
+        screen_name : str
+            Name to supply for self.screen.
+        default_screen : zipline.pipeline.term.Term
+            Term to use as a screen if self.screen is None.
+        """
+        return TermGraph(self._prepare_graph_terms())
```

Comment:
Good catch.  The codepath that calls this isn't tested because it relies on having the `dot` program installed.  I'll see if I can at least throw in a smoketest for this.